common/core/web/keyboard-processor/src/keyboards/activeLayout.ts

```diff
@@ -652,15 +652,11 @@ namespace com.keyman.keyboards {
         let layer=layers[n];
         let rows=layer['row'];
         let i: number;
-        for(i=rows.length; i>0; i--) {
-          if(rows[i-1]['key'].length > 0) {
-            break;
+        for(i=rows.length-1; i>=0; i--) {
+          if(!Array.isArray(rows[i]['key'])) {
+            rows.splice(i, 1)
```

Comment:
The previous implementation wasn't removing an empty row if it was the top one (rows[0])

Reply:
Should also include the `.length` check, as `Array.isArray([]) == true`.  What you put here does catch `undefined` cases (which was missing before), but not empty-array cases (which were caught before).

Reply:
Gotcha. I've updated to include both checks